covid_graphs/covid_graphs/scripts/server.py

```diff
@@ -37,42 +37,54 @@ def run_server(data_dir: Path) -> None:
     """Creates and runs a flask server."""
     server = Flask(__name__, template_folder=str(CURRENT_DIR))
 
-    @server.route("/")
-    def home():
-        return render_template("index.html")
-
-    @server.route("/covid19/normal/")
-    def covid19_normal_redirect():
-        return redirect(url_for("covid19_all_predictions"))
-
-    @server.route("/covid19/predictions/")
-    def covid19_predictions_redirect():
-        return redirect(url_for("covid19_single_predictions"))
+    _setup_server(server, data_dir)
 
     if platform_supports_inotify:
         i = adapters.InotifyTree(
             str(data_dir), mask=(constants.IN_MODIFY | constants.IN_DELETE | constants.IN_CREATE)
         )
-        p = Process(target=_run_flask_server, kwargs=dict(server=server, data_dir=data_dir))
+        p = Process(target=_run_flask_server, kwargs=dict(server=server))
         p.start()
         while True:
             events = list(i.event_gen(yield_nones=False, timeout_s=1))
             if len(events) != 0:
                 print("Data changed. Restarting server...")
                 p.terminate()
                 p.join()
-                p = Process(target=_run_flask_server, kwargs=dict(server=server, data_dir=data_dir))
+                p = Process(target=_run_flask_server, kwargs=dict(server=server))
                 p.start()
                 print("Server restarted.")
             sleep(10)
     else:
-        _run_flask_server(server=server, data_dir=data_dir)
+        _run_flask_server(server=server)
+
+
+def uswgi_server(data_dir: Path) -> Flask:
```

Comment:
This function happens to be used by `uswgi`, but that doesn't mean that we need to call it that.

I think this file should be factored differently.
```suggestion
def setup_server(data_dir: Path) -> Flask:
server = Flask(__name__, template_folder=str(CURRENT_DIR))
... setting up all @server.route's ...
return server
```

That is, change `_setup_server` into a public function.

Reply:
Refactored in new commits.